types/signer.go

```diff
@@ -31,11 +31,11 @@ var (
 
 type BeaconSigner interface {
 	// SignBeaconObject returns signature and root.
-	SignBeaconObject(obj ssz.HashRoot, domain spec.Domain, pk []byte) (Signature, []byte, error)
+	SignBeaconObject(obj ssz.HashRoot, domain spec.Domain, pk []byte, role BeaconRole) (Signature, []byte, error)
```

Comment:
why do we need to pass role?

Reply:
we want to be sure that there is no slashing event before signing. 

here example:
https://github.com/bloxapp/ssv/blob/6b4cde4cc1774d3b07e605d1782bd75b4e0848e4/ekm/eth_key_manager_signer.go#L89

the slashing check only for attester & proposer roles

Reply:
in theory we can check the slashing protection after we signed the object